File: ZoneSentry/Models/ResidentialComplex.cs
namespace ZoneSentry.Models;

public class ResidentialComplex
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public string City { get; set; }
    
    public ConstructionCompany ConstructionCompany { get; set; }
    public int ConstructionCompanyId { get; set; }
    
    public List<House> Houses { get; set; }
}

public class ResidentialComplexDTO
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public string City { get; set; }
    
    public int ConstructionCompanyId { get; set; }
    
    public List<int> Houses { get; set; }
}

public class ResidentialComplexCreate
{
    public string Name { get; set; }
    public string City { get; set; }
}

File: ZoneSentry/ClientApp/src/api/models/ResidentialComplex.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ConstructionCompany } from './ConstructionCompany';
import type { House } from './House';

export type ResidentialComplex = {
    id?: number;
    name?: string | null;
    city?: string | null;
    constructionCompany?: ConstructionCompany;
    constructionCompanyId?: number;
    houses?: Array<House> | null;
};